plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -802,85 +823,98 @@ private function generate_select_clause_for_props( $table_alias, $props ) {
-				),
-				$row
+			$db_updates[] = array(
+				'table'        => self::get_orders_table_name(),
+				'pk_table'     => true,
+				'data'         => $row['data'],
+				'format'       => $row['format'],
+				'where'        => $is_new_record ? null : array( 'id' => $order_id ),
+				'where_format' => $is_new_record ? null : '%d',
 			);
 		}
 
 		// wc_order_operational_data.
-		$row = $this->get_db_row_from_order_changes(
-			array_merge(
-				$changes,
-				// XXX: manually persist some of the properties until the datastore/property design is finalized.
-				array(
-					'stock_reduced'                => $this->get_stock_reduced( $order ),
-					'download_permissions_granted' => $this->get_download_permissions_granted( $order ),
-					'new_order_email_sent'         => $this->get_email_sent( $order ),
-					'recorded_sales'               => $this->get_recorded_sales( $order ),
-					'recorded_coupon_usage_counts' => $this->get_recorded_coupon_usage_counts( $order ),
-				)
-			),
-			$this->operational_data_column_mapping
-		);
+		$row = $this->get_db_row_from_order_changes( $changes, $this->operational_data_column_mapping );
 		if ( $row ) {
-			$db_updates[] = array_merge(
-				array(
-					'table'        => self::get_operational_data_table_name(),
-					'where'        => array( 'order_id' => $order->get_id() ),
-					'where_format' => '%d',
-				),
-				$row
+			$db_updates[] = array(
+				'table'        => self::get_operational_data_table_name(),
+				'data'         => $row['data'],
+				'format'       => $row['format'],
+				'where'        => $is_new_record ? null : array( 'order_id' => $order_id ),
+				'where_format' => $is_new_record ? null : '%d',
 			);
 		}
 
 		// wc_order_addresses.
 		foreach ( array( 'billing', 'shipping' ) as $address_type ) {
 			$row = $this->get_db_row_from_order_changes( $changes, $this->{$address_type . '_address_column_mapping'} );
 
+			$row['data']['address_type'] = $address_type;
+			$row['format'][]             = '%s';
+
 			if ( $row ) {
-				$db_updates[] = array_merge(
-					array(
-						'table'        => self::get_addresses_table_name(),
-						'where'        => array(
-							'order_id'     => $order->get_id(),
-							'address_type' => $address_type,
-						),
-						'where_format' => array( '%d', '%s' ),
+				$db_updates[] = array(
+					'table'        => self::get_addresses_table_name(),
+					'data'         => $row['data'],
+					'format'       => $row['format'],
+					'where'        => $is_new_record ? null : array(
+						'order_id'     => $order_id,
+						'address_type' => $address_type,
 					),
-					$row
+					'where_format' => $is_new_record ? null : array( '%d', '%s' ),
 				);
 			}
 		}
 
 		// Persist changes.
 		foreach ( $db_updates as $update ) {
-			$wpdb->update(
-				$update['table'],
-				$update['row'],
-				$update['where'],
-				array_values( $update['format'] ),
-				$update['where_format']
-			);
+			$is_pk_table = ! empty( $update['pk_table'] );
```

Comment:
as mentioned in another comment, it would be good to insert in posts table now and use that ID

Reply:
✔️